mobile_app_connector/models/app_hub.py

```diff
@@ -71,12 +77,36 @@ def mobile_get_message(self, partner_id, **pagination):
             'product.product': products,
             'correspondence': letters,
         }
+        unpaid_data = {
+            'recurring.contract': unpaid
+        }
         # TODO handle pagination properly
         limit = int(pagination.get('limit', 1000))
         messages = available_tiles[:limit].render_tile(tile_data)
+        msg_tmp = self.env['mobile.app.tile'].search([
+            ('name', '=', 'Awaiting payment')
+        ])[:limit].render_tile(unpaid_data)
+        for msg in msg_tmp:
+            msg['Child']['SupporterId'] = self.env.user.id
+            msg['Child']['SupporterGroupId'] = self.env.user.id
+        messages.extend(msg_tmp)
         messages.extend(self._fetch_wordpress_tiles(**pagination))
         res = self._construct_hub_message(
             partner_id, messages, children, **pagination)
+
+        # Handle children with awaiting payment
+        if unpaid_children is not None:
+            unpaid_dict = unpaid_children.get_app_json(multi=True)
+            if unpaid_dict.get('Children') is not None:
+                unpaid_dict['UnpaidChildren'] = unpaid_dict['Children']
+                del unpaid_dict['Children']
+                for i in range(len(unpaid_dict['UnpaidChildren'])):
```

Comment:
Isn't it handled already at line `89`?

Reply:
No, it is two different things. At line `89`, messages containing tile info are updated. At line `100` and afterward, it updates objects that will be sent as **HomeResponse**, which also contains information for "awaiting payment" tiles but not the same.
But I think their should be a solution to avoid both, but I cannot find it...